### PR TITLE
feat: add key_id & algorithm to kmstool-enclave-cli decrypt

### PR DESCRIPTION
**Pull request by emperorhan**
Modified to accept key-id and encryption-algorithm in aws_kms_decrypt_blocking API so that kmstool-enclave-cli can support kms rsa decryption.

### CODE REVIEW

File: include/aws/nitro_enclaves/kms.h [hunk 1]
```diff
@@ -971,14 +971,18 @@ void aws_nitro_enclaves_kms_client_destroy(struct aws_nitro_enclaves_kms_client
...
  * Calling it from a non-enclave environment will fail.
  *
- * @param[in]   client      The AWS KMS client to use for calling the API.
- * @param[in]   ciphertext  The ciphertext to decrypt.
- * @param[out]  plaintext   The plaintext output of the call. Should be an empty, but non-null aws_byte_buf.
- * @return                  Returns AWS_OP_SUCCESS if the call succeeds and plaintext is populated.
+ * @param[in]   client                  The AWS KMS client to use for calling the API.
+ * @param[in]   key_id                  The ARN or alias of AWS KMS CMK used to encrypt the data key.
+ * @param[in]   encryption_algorithm    The encryption algorithm that will be used to decrypt the ciphertext.
+ * @param[in]   ciphertext              The ciphertext to decrypt.
```
Comment (shtaked) [inline include/aws/nitro_enclaves/kms.h:977]: Please add some notes that `ciphertext` may already contain key id as a metadata for symmetric keys and in this case `key_id` and `encryption_algorithm` may be `NULL`

File: bin/kmstool-enclave-cli/main.c [hunk 2]
```diff
@@ -56,6 +58,11 @@ static void s_usage(int exit_code) {
     fprintf(stderr, "    --aws-secret-access-key SECRET_ACCESS_KEY: AWS secret access key\n");
     fprintf(stderr, "    --aws-session-token SESSION_TOKEN: Session token associated with the access key ID\n");
     fprintf(stderr, "    --ciphertext CIPHERTEXT: base64-encoded ciphertext that need to decrypt\n");
+    fprintf(stderr, "    --key-id KEY_ID: decrypt key id\n");
```
Comment (shtaked) [inline bin/kmstool-enclave-cli/main.c:61]: Same here: probably worth to mention that it's optional for symmetric keys

File: bin/kmstool-enclave-cli/main.c [hunk 3]
```diff
@@ -68,6 +75,8 @@ static struct aws_cli_option s_long_options[] = {
     {"aws-secret-access-key", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 's'},
     {"aws-session-token", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 't'},
     {"ciphertext", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 'c'},
+    {"key-id", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 'K'},
```
Comment (shtaked) [inline bin/kmstool-enclave-cli/main.c:78]: may be small `k` for consistency?

File: bin/kmstool-enclave-cli/main.c [hunk 4]
```diff
@@ -142,6 +159,14 @@ static void s_parse_options(int argc, char **argv, struct app_ctx *ctx) {
         exit(1);
     }
 
+    // if key id is set check encryption algorithm is exists
+    if (ctx->key_id != NULL) {
+        if (ctx->encryption_algorithm == NULL) {
+            fprintf(stderr, "--encryption-algorithm must be set at key-id exists\n");
```
Comment (shtaked) [inline bin/kmstool-enclave-cli/main.c:165]:
```suggestion
            fprintf(stderr, "--encryption-algorithm must be set if key-id exists\n");
```

**Review by shtaked — CHANGES_REQUESTED** [review]

File: bin/kmstool-enclave-cli/main.c (hunk 3 above)
Comment (emperorhan) [reply bin/kmstool-enclave-cli/main.c:78]: Since small `k` is already assigned to `aws-access-key-id`, I set it to large `K`, do I need to modify it?

Comment (shtaked) [reply bin/kmstool-enclave-cli/main.c:78]: sorry, didn't notice it, yes, then let's keep it as `K`

**Review by shtaked — DISMISSED** [review]

**Comment by shtaked** [comment]
With absence of integration tests, pending a manual check on our side

**Comment by shtaked** [comment]
Verified manually

**Comment by petreeftime** [comment]
Sorry for taking this long with the review. LGTM, but could you please add a `Signed-off-by` in the commits? `git rebase --signoff HEAD~2` should do this. Afterwards, I will merge.

**Comment by aceeric** [comment]
Folks - did this get reverted? Looking at https://github.com/aws/aws-nitro-enclaves-sdk-c/blob/main/bin/kmstool-enclave-cli/main.c, the default parse case for `decrypt` does not honor `'K' `and so at a minimum, `--key-id` is not accepted in a decrypt operation... And in fact in my testing, I'm getting an error if I try to specify a `--key-id 55f949e7-7bac-4c8b-9a11-324bbe281dd9`:
```
Unknown option: 55f949e7-7bac-4c8b-9a11-324bbe281dd9
```

**Pull request merged**